Makefile

```
@@ -44,31 +50,57 @@ endif
 usb_host=$(shell yarn run -s discovery find -i 169.254)
 
 # install all project dependencies
+# may be run with -j
 .PHONY: setup
 setup: setup-js setup-py
 
-# front-end dependecies handled by yarn
-.PHONY: setup-js
-setup-js:
+.PHONY: setup-js-root
+setup-js-root: setup-js-globals
 	yarn config set network-timeout 60000
 	yarn
-	$(MAKE) -C $(APP_SHELL_DIR) setup
+
+.PHONY: setup-app-shell-odd
+setup-app-shell-odd: setup-js-root
 	$(MAKE) -C $(APP_SHELL_ODD_DIR) setup
+
+.PHONY: setup-app-shell
+setup-app-shell: setup-js-root
+	$(MAKE) -C $(APP_SHELL_DIR) setup
+
+.PHONY: setup-js-shared-data
+setup-js-shared-data: setup-js-root
 	$(MAKE) -C $(SHARED_DATA_DIR) setup-js
 
-PYTHON_SETUP_TARGETS := $(addsuffix -py-setup, $(PYTHON_DIRS))
+# front-end dependecies handled by yarn
+.PHONY: setup-js
+setup-js: setup-js-globals setup-js-root setup-app-shell setup-app-shell-odd setup-js-shared-data
 
-.PHONY: setup-py
-setup-py:
+# this is the source of truth for pipenv version
+.PHONY: setup-python-globals
+setup-python-globals:
 	$(OT_PYTHON) -m pip install pipenv==2021.5.29
-	$(MAKE) $(PYTHON_SETUP_TARGETS)
 
+# this is the source of truth for yarn version
+.PHONY: setup-js-globals
+setup-js-globals:
+	npm i -g yarn@1
```

Thanks to #11889, we can defer this work to the [corepack](https://github.com/nodejs/corepack) builtin of Node.js

```suggestion
	corepack enable
```

This tells Node to fetch the proper version of yarn (spec'd in `package.json`) on first call